content/snippets/capacity-planning.md

```diff
@@ -0,0 +1,97 @@
+# Capacity Planning
+
+
+Even the most minimally functional `rippled` server must contain the most recently validated Ledger versions to submit transactions to the network and verify the integrity of the XRP Ledger. Beyond these, consider the following server requirements:
+
+- Handling ever-increasing transaction volume
+- Servicing transaction reporting information to clients
+- Maintaining varying amounts of historical data
+
+Capacity planning for `rippled` involves two factors:
+
+- The [configuration settings](#configuration-settings) that affect resource utilization
+- The [network and hardware] (#network and hardware) requirements to achieve consistent, good performance across the XRP Ledger network
+
+## Configuration Settings
+
+Ripple recommends the following guidelines to improve performance. You can set the following parameters in the `rippled.cfg` file to improve performance for your `rippled` server.
+
+### Node Size
+
+The `node_size` parameter determines the size of database caches. Ripple recommends you always set it to `huge` for production servers. Increased cache size requires less disk I/O and allows `rippled` to improve performance. The trade-off to improving performance better is that memory requirements increase.
+
+### Node DB Type
+
+The `type` field in the `node_db` section of the `rippled.cfg` file sets the type of key-value store that `rippled` uses to persist the XRP Ledger in the Ledger Store. You can set the value to either `rocksdb` or `nudb`. For setting the store that `rippled` uses to persist history shards, if you are participating in history sharding, see [History Sharding](XREF: concept-history-sharding.md#shard-store-configuration).
+
+#### RocksDB vs NuDB
+RocksDB requires approximately one-third less disk storage than NuDB and provides a corresponding improvement in I/O latency. However, this comes at a cost of increased memory utilization as storage size grows. NuDB, on the other hand, has nearly constant performance and memory footprint regardless of storage.
+
+For `rippled` servers that operate as validators, which keep only a few days' worth of data or less, Ripple recommends using RocksDB. For all other uses, Ripple recommends users configure NuDB for the Ledger Store. NuDB has no performance-related configuration options. But tuning parameters are available for RocksDB that have been used to help achieve maximum tested transaction processing throughput, and are as follows:
+
+```
+[node_db]
+type=rocksdb
+open_files=512
+file_size_mb=64
+file_size_mult=2
+filter_bits=12
+cache_mb=512
+path={path_to_ledger_store}
+```
+
+### Historical Data
+
+The amount of historical data kept online, as set in the `online_delete` and  `advisory_delete` field, is a major contributor to required storage space. Currently, about 12GB of data is stored per day. However, you should expect this amount to grow as transaction volume increases across the XRP Ledger network.
```

Comment:
Received some feedback in the IA interviews that folks want to know more about how to use the `online_delete` feature -- beyond what is currently covered in the `rippled.cfg` inline doc. Is there anything more to say on this topic in the context of Capacity Planning? Is our recommendation to stick with the default `2000`? If you leave it set at `2000`, does that result in the 12GB stored per day? If you leave `advisory_delete` set to `0`, it sounds like the `online_delete` can happen without an external admin command - does that mean that once you hit 2,000 stored ledgers, the oldest ledgers will automatically be deleted from the Ledger Store? Perhaps this full discussion is outside the scope of this doc -- but thought I'd mention it since I heard a request for more on this topic.

Reply:
@mtrippled we're trying to deprecate that wiki, hence the red banner ;)

@jhaaaa @mtrippled we have a few stories for improving our config documentation beyond the example file in the `rippled` repo. I've added a specific bullet point for `online_delete` recommendations to this one: https://ripplelabs.atlassian.net/browse/DOC-1313 (Ripple employee access only). If we need to, we'll create a separate story just for `online_delete` configuration suggestions.